hedera-node/hedera-app/src/main/java/com/hedera/node/app/workflows/handle/HandleContextImpl.java

```diff
@@ -392,68 +398,67 @@ public <T> T dispatchPrecedingTransaction(
-    }
-
-    @Override
-    @NonNull
-    public <T> T dispatchRemovableChildTransaction(
-            @NonNull final TransactionBody txBody, @NonNull final Class<T> recordBuilderClass) {
+            @NonNull final Predicate<Key> callback) {
         final var childRecordBuilder = recordListBuilder.addRemovableChild(configuration());
-        return dispatchChildTransaction(txBody, verifier, childRecordBuilder, recordBuilderClass);
+        return doDispatchChildTransaction(txBody, childRecordBuilder, recordBuilderClass, callback);
     }
 
     @NonNull
-    private <T> T dispatchChildTransaction(
+    private <T> T doDispatchChildTransaction(
             @NonNull final TransactionBody txBody,
-            @NonNull final HandleContextVerifier childVerifier,
             @NonNull final SingleTransactionRecordBuilderImpl childRecordBuilder,
-            @NonNull final Class<T> recordBuilderClass) {
+            @NonNull final Class<T> recordBuilderClass,
+            @NonNull final Predicate<Key> callback) {
+        requireNonNull(txBody, "txBody must not be null");
+        requireNonNull(recordBuilderClass, "recordBuilderClass must not be null");
+        requireNonNull(callback, "callback must not be null");
+
         if (category == PRECEDING) {
             throw new IllegalArgumentException("A preceding transaction cannot have child transactions");
         }
 
-        // create a savepoint
-        stack.createSavepoint();
-
         // run the child-transaction
-        dispatchSyntheticTxn(txBody, CHILD, childVerifier, childRecordBuilder);
-
-        // rollback if the child-transaction failed
-        if (childRecordBuilder.status() != ResponseCodeEnum.OK) {
-            stack.rollback();
-        }
+        dispatchSyntheticTxn(txBody, CHILD, childRecordBuilder, callback);
 
         return castRecordBuilder(childRecordBuilder, recordBuilderClass);
     }
 
     private void dispatchSyntheticTxn(
             @NonNull final TransactionBody txBody,
             @NonNull final TransactionCategory childCategory,
-            @NonNull final HandleContextVerifier childVerifier,
-            @NonNull final SingleTransactionRecordBuilderImpl childRecordBuilder) {
+            @NonNull final SingleTransactionRecordBuilderImpl childRecordBuilder,
+            @NonNull final Predicate<Key> callback) {
+        // Initialize record builder list
+        final var bodyBytes = TransactionBody.PROTOBUF.toBytes(txBody);
+        final var signedTransaction =
+                SignedTransaction.newBuilder().bodyBytes(bodyBytes).build();
+        final var signedTransactionBytes = SignedTransaction.PROTOBUF.toBytes(signedTransaction);
+        final var transaction = Transaction.newBuilder()
+                .signedTransactionBytes(signedTransactionBytes)
+                .build();
+        final var transactionBytes =
+                Bytes.wrap(PbjConverter.fromPbj(transaction).toByteArray());
```

Comment:
Not critical, but shouldn't the transaction bytes be created using `Transaction.PROTOBUF.toBytes(transaction)` instead?